File: src/main/java/leetcode/topinterview/LeetCode26.java
package leetcode.topinterview;

/**
 * @author wujingxinit@outlook.com
 * @date 2022/9/21 01:08
 */
public class LeetCode26 {

    public int removeDuplicates(int[] nums) {
        int x = 1;
        for (int i = x; i < nums.length; i++) {
            if (nums[i] > nums[x - 1]) {
                nums[x++] = nums[i];
            }
        }
        return x;
    }
}

File: go/src/leetcode/topinterview/LeetCode26.go
package topinterview

func removeDuplicates(nums []int) int {
	if len(nums) == 0 {
		return 0
	}
	left, right := 0, 1
	for right < len(nums) {
		if nums[right] <= nums[left] {
			right++
		} else {
			left++
			nums[left], nums[right] = nums[right], nums[left]
		}
	}
	return left + 1
}
